agents/heuristics_resource/fence_heuristics_resource.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/libexec/platform-python -tt
+
+import io
+import re
+import subprocess
+import shlex
+import sys, stat
+import logging
+import os
+import atexit
+import time
+sys.path.append("/usr/share/fence")
+from fencing import fail_usage, run_command, fence_action, all_opt
+from fencing import atexit_handler, check_input, process_input, show_docs
+from fencing import run_delay
+
+def heuristics_resource(con, options):
+
+	if options["--action"] == "on":
+		return True
+
+	if not "--resource" in options or options["--resource"] == "":
+		logging.error("resource parameter required")
+		return False
+
+	crm_resource_path = options["--crm-resource-path"]
+	resource = options["--resource"]
+	standby_wait = int(options["--standby-wait"])
+	p = None
+	cmd = "%s -r %s -W" % (crm_resource_path, resource)
+	search_str = re.compile(r"\s%s$" % os.uname()[1])
```

Comment:
as pacemaker doesn't necessarily use uname as name for the node it would be preferable here to use `crm_node --name` or something rather.

Reply:
Fixed to use `crm_node --name`